packages/ethpm-v3/lib/utils.js

```diff
@@ -0,0 +1,301 @@
+  const sources = await resolveSources(
+    sourcePaths,
+    options.contracts_directory,
+    ethpm
+  );
+  return {
+    resolvedSources: sources,
+    resolvedContractTypes: parsedArtifacts.contractTypes,
+    resolvedDeployments: parsedArtifacts.deployments,
+    resolvedCompilers: parsedArtifacts.compilers
+  };
+}
+
+async function resolveEnsName(address, provider, options) {
+  let resolvedAddress;
+  const w3 = new Web3(provider);
+  const connectedChainId = await w3.eth.net.getId();
+  const supportedChainIds = [1, 3, 4, 5];
+  if (!supportedChainIds.includes(connectedChainId)) {
+    throw new TruffleError(
+      "Invalid ethPM uri. ENS is only supported on mainnet, ropsten, rinkeby & goerli."
+    );
+  }
+  if (options.ens.enabled === false) {
+    throw new TruffleError(
+      "ENS must be enabled in truffle-config to use ENS names."
+    );
+  }
+  try {
+    var ens = new ENS(provider);
+    resolvedAddress = await ens.resolver(address).addr();
+  } catch (err) {
+    throw new TruffleError(
+      `Unable to resolve ENS name: ${address}. Reason: ${err.message}`
+    );
+  }
+  return resolvedAddress;
+}
+
+async function resolveEthpmUri(options, provider) {
+  let targetRegistry;
+  var targetProvider = provider;
+  const ethpmUri = new EthpmURI(options.packageIdentifier);
+  const w3 = new Web3(targetProvider);
+  const connectedChainId = await w3.eth.net.getId();
+
+  // update provider if it doesn't match chain id in ethpm uri
+  if (
+    connectedChainId != ethpmUri.chainId &&
+    options.ethpm.registry.network != "development"
+  ) {
+    const targetNetwork = SUPPORTED_CHAIN_IDS[ethpmUri.chainId];
+    if (typeof targetNetwork === "undefined") {
+      throw new TruffleError(
+        `Invalid ethpm uri. Unsupported chain id: ${ethpmUri.chainId}`
+      );
+    }
+    targetProvider = options.networks[targetNetwork].provider();
+  }
+
+  // Resolve ENS names in ethpm uri
+  if (!isAddress(ethpmUri.address)) {
+    targetRegistry = await resolveEnsName(
+      ethpmUri.address,
+      targetProvider,
+      options
+    );
+  } else {
+    targetRegistry = ethpmUri.address;
+  }
+
+  if (!ethpmUri.packageName) {
+    throw new TruffleError(
+      "Invalid ethpm uri. To install from an ethpm uri, you must at least provide the package name to install."
+    );
+  }
+
+  return {
+    targetPackageName: ethpmUri.packageName,
+    targetVersion: ethpmUri.version,
+    targetNetworkId: ethpmUri.chainId,
+    targetRegistry: targetRegistry,
+    targetProvider: targetProvider
+  };
+}
+
+function fetchInstalledBuildDependencies(workingDirectory) {
+  let ethpmLock;
+  try {
+    ethpmLock = JSON.parse(
+      fs.readFileSync(
+        path.join(workingDirectory, "_ethpm_packages", "ethpm.lock"),
+        "utf8"
+      )
+    );
+  } catch (err) {
+    return {};
+  }
+  const installedBuildDependencies = {};
+  Object.keys(ethpmLock).forEach(function (key, _) {
```

Comment:
You can just omit the 2nd argument (`_`) in this case.

An arrow function would be slickest.
```
Object.keys(ethpmLock).forEach(key => installedBuildDependencies[key] = ethpmLock[key].resolved_uri)
```